packages/grid/x-data-grid/src/models/events/gridEventLookup.ts

```diff
@@ -27,6 +27,22 @@ export interface GridRowEventLookup {
   rowDoubleClick: { params: GridRowParams; event: React.MouseEvent<HTMLElement> };
   rowMouseEnter: { params: GridRowParams; event: React.MouseEvent<HTMLElement> };
   rowMouseLeave: { params: GridRowParams; event: React.MouseEvent<HTMLElement> };
+  rowDragStart: {
```

Comment:
Those new events are pro-only and can be defined on the module augmentation of `x-data-grid-pro` (search for `GridEventLookupPro`)

Reply:
It same reason as https://github.com/mui/mui-x/pull/4034#discussion_r829908169, unfortunately.